imports/plugins/included/taxes-taxcloud/client/containers/taxCloudSettingsFormContainer.js

```diff
@@ -0,0 +1,117 @@
+    * @property {String} packageName - the name of this tax provider in lowercase.
+    * @since 1.5.2
+    * @return {null} - returns nothing
+    */
+  handleSubmit(event, changedInfo, targetField) {
+    if (!changedInfo.isValid) {
+      return;
+    }
+    Meteor.call("package/update", this.props.packageName, targetField, changedInfo.doc.settings.taxcloud, (error) => {
+      if (error) {
+        Alerts.toast(
+          i18next.t("admin.update.updateFailed", { defaultValue: "Failed to update TaxCloud settings." }),
+          "error"
+        );
+        return;
+      }
+      Alerts.toast(
+        i18next.t("admin.update.updateSucceeded", { defaultValue: "TaxCloud settings updated." }),
+        "success"
+      );
+    });
+  }
+
+  render() {
+    const providerName = this.props.providerName;
+    const packageSettings = this.props.settings;
+    const shownFields = {
+      [`settings.${providerName}.apiKey`]: TaxCloudPackageConfig._schema[`settings.${providerName}.apiKey`],
+      [`settings.${providerName}.apiLoginId`]: TaxCloudPackageConfig._schema[`settings.${providerName}.apiLoginId`]
+    };
+    const hiddenFields = [
+      `settings.${providerName}.enabled`,
+      `settings.${providerName}.refreshPeriod`,
+      `settings.${providerName}.taxCodeUrl`
+    ];
+
+    return (
+      <div>
+        {!packageSettings.taxcloud.apiLoginId &&
+          <div className="alert alert-info">
+            <span data-i18n="admin.taxSettings.taxcloudCredentials">Add API Login ID to enable</span>
+            <a href="https://www.taxcloud.com/" target="_blank">TaxCloud</a>
+          </div>
+        }
+        <TaxCloudSettingsForm
+          schema={TaxCloudPackageConfig}
+          doc={{ settings: packageSettings }}
+          docPath={`settings.${providerName}`}
+          name={`settings.${providerName}`}
+          fields={shownFields}
+          hideFields={hiddenFields}
+          handleSubmit={this.handleSubmit}
+        />
+      </div>
+    );
+  }
+}
+
+/**
+  * @name TaxCloudSettingsFormContainer propTypes
+  * @type {propTypes}
+  * @param {Object} props - React PropTypes
+  * @property {String} packageName - the value of the "name" field in the TaxCloud Package.
+  * @property {String} providerName - the name of this provider in lowercase i.e "taxcloud".
+  * @property {Object} settings - the value of the "settings" field in the TaxCloud Package.
+  * @return {Array} React propTypes
+  */
+TaxCloudSettingsFormContainer.propTypes = {
+  packageName: PropTypes.string,
+  providerName: PropTypes.string,
+  settings: PropTypes.object
+};
+
+const composer = (props, onData) => {
+  const shopId = Reaction.getShopId();
+  const packageSub = Meteor.subscribe("Packages", shopId);
+  if (packageSub.ready()) {
+    const providerName = "taxcloud";
+    const packageName = "taxes-taxcloud";
+    const packageData = Packages.findOne({ name: packageName, shopId });
```

Comment:
We should be using `getPackageSettings` again as I had mentioned before

Reply:
For this section, I actually thought a lot about using `getPackageSettings()`. But I eventually thought that "if I wait for `packageSub.ready()`, why not use `Packages.findOne()` after the subscription is ready?"

Reply:
Done.